packages/core/admin/server/src/services/homepage.ts

```diff
@@ -1,129 +1,212 @@
-   * more precise queries than exact key matches, in order to make as few queries as possible.
-   */
-  const coreStore = strapi.db.query('strapi::core-store');
-
-  return {
-    async getRecentUpdates(): Promise<GetRecentDocuments.Response['data']> {
-      // Get all the content types the user has permissions to read
-      const readPermissions = await permissionService.findMany({
-        where: {
-          role: { users: { id: strapi.requestContext.get()?.state?.user.id } },
-          action: 'plugin::content-manager.explorer.read',
+  type ContentTypeConfiguration = {
+    uid: RecentDocument['model'];
+    settings: { mainField: string };
+  };
+  const getConfiguration = async (
+    contentTypeUids: RecentDocument['model'][]
+  ): Promise<ContentTypeConfiguration[]> => {
+    /**
+     * Don't use the strapi.store util because we need to make
+     * more precise queries than exact key matches, in order to make as few queries as possible.
+     */
+    const coreStore = strapi.db.query('strapi::core-store');
+    const rawConfigurations = await coreStore.findMany({
+      where: {
+        key: {
+          $in: contentTypeUids.map(
+            (contentType) => `plugin_content_manager_configuration_content_types::${contentType}`
+          ),
         },
-      });
-      const allowedContentTypeNames = readPermissions
-        .map((permission) => permission.subject)
-        .filter(Boolean) as UID.ContentType[];
+      },
+    });
 
-      // Fetch the configuration for each content type in a single query
-      const rawConfigurations = await coreStore.findMany({
-        where: {
-          key: {
-            $in: allowedContentTypeNames.map(
-              (contentType) => `plugin_content_manager_configuration_content_types::${contentType}`
-            ),
-          },
-        },
-      });
-      const configurations = rawConfigurations.map((rawConfiguration) => {
-        return JSON.parse(rawConfiguration.value);
-      });
+    return rawConfigurations.map((rawConfiguration) => {
+      return JSON.parse(rawConfiguration.value);
+    });
+  };
 
-      const recentDocuments = await Promise.all(
-        allowedContentTypeNames.map(async (contentTypeName) => {
-          const configuration = configurations.find((config) => config.uid === contentTypeName);
-          const contentType = strapi.contentType(contentTypeName);
-          const fields = ['documentId', 'updatedAt'];
-
-          // Add fields required to get the status if D&P is enabled
-          const hasDraftAndPublish = contentTypes.hasDraftAndPublish(contentType);
-          if (hasDraftAndPublish) {
-            fields.push('publishedAt');
-          }
+  const getPermittedContentTypes = async () => {
+    const readPermissions = await permissionService.findMany({
+      where: {
+        role: { users: { id: strapi.requestContext.get()?.state?.user.id } },
+        action: 'plugin::content-manager.explorer.read',
+      },
+    });
```

Comment:
how does this work for super admins? do we also store super admin permissions in the db?

Reply:
I think so. I've been testing with my super admin account JWT

Reply:
Not sure 🤔 @remidej any idea? Otherwise I'll look into it.

Reply:
oops didn't see your comment 🙃 ✅